runtime/gc_ctrl.c

```diff
@@ -392,16 +390,30 @@ CAMLprim value caml_runtime_variant (value unit)
 #endif
 }
 
-extern int caml_parser_trace;
-
 CAMLprim value caml_runtime_parameters (value unit)
 {
 #define F_Z ARCH_INTNAT_PRINTF_FORMAT
 #define F_S ARCH_SIZET_PRINTF_FORMAT
 
-  CAMLassert (unit == Val_unit);
-  /* TODO KC */
-  return caml_alloc_sprintf ("caml_runtime_parameters not implemented: %d", 0);
+    CAMLassert (unit == Val_unit);
+    return caml_alloc_sprintf
+        ("b=%d,e=%d,l=%"F_Z"u,M=%"F_Z"u,m=%"F_Z"u,n=%"F_Z"u,o=%"F_Z"u,p=%d,"
+         "s=%"F_S"u,v=%"F_Z"u,V=%"F_Z"u,W=%"F_Z"u",
+         /* b */ (int) Caml_state->backtrace_active,
+         /* e */ caml_runtime_events_ring_size_words,
+         /* l */ caml_max_stack_wsize,
+         /* M */ caml_custom_major_ratio,
+         /* m */ caml_custom_minor_ratio,
+         /* n */ caml_custom_minor_max_bsz,
+         /* o */ caml_percent_free,
+         /* p */ Caml_state->parser_trace,
+         /* s */ Caml_state->minor_heap_wsz,
+         /* v */ caml_verb_gc,
+         /* V */ caml_gc_verify_heap,
+         /* W */ caml_runtime_warnings
+         );
```

Comment:
Spacing nit, and I think the comment about "R" is worth preserving from 4.14
```suggestion
  CAMLassert (unit == Val_unit);
  return caml_alloc_sprintf
      ("b=%d,e=%d,l=%"F_Z"u,M=%"F_Z"u,m=%"F_Z"u,n=%"F_Z"u,o=%"F_Z"u,p=%d,"
       "s=%"F_S"u,v=%"F_Z"u,V=%"F_Z"u,W=%"F_Z"u",
       /* b */ (int) Caml_state->backtrace_active,
       /* e */ caml_runtime_events_ring_size_words,
       /* l */ caml_max_stack_wsize,
       /* M */ caml_custom_major_ratio,
       /* m */ caml_custom_minor_ratio,
       /* n */ caml_custom_minor_max_bsz,
       /* o */ caml_percent_free,
       /* p */ Caml_state->parser_trace,
       /* R */ /* missing */
       /* s */ Caml_state->minor_heap_wsz,
       /* v */ caml_verb_gc,
       /* V */ caml_gc_verify_heap,
       /* W */ caml_runtime_warnings
       );
```